app/controllers/routes_controller.rb

```diff
@@ -11,9 +11,13 @@ def show
 
   def update
     route_details = @request_data[:route]
-    @route = Route.find_or_initialize_by(incoming_path: route_details.delete(:incoming_path))
-    status_code = @route.new_record? ? 201 : 200
-    @route.update_attributes(route_details) || status_code = 422
+    @route = Route.new(route_details)
+    if @route.valid?
```

Comment:
Maybe setting a new `attr_accessor` on `Route` for something like `for_upsert` would work, then you could keep the uniqueness check.

```ruby
@route = Route.new(route_details)
@route.for_upsert = true
if @route.valid?
  # snip
```

Then in `route.rb`:

```ruby
validates :incoming_path, uniqueness: true, unless: :for_upsert
```

I'm not entirely convinced that helps much by re-enabling the uniqueness check.